### PR TITLE
respect aliases SKIP when installing environments

### LINKED ISSUE #2478: SKIP skips invocation by alias but doesn't skip "Installing environment"
### search tried in the issue tracker

SKIP alias

### describe your issue

When I run 

```sh
SKIP=pylint-alias2 git commit
```

the hook aliased as `pylint-alias2` is properly skipped, however, its environment is still initialized. I would expect that it also skips the lengthy initialization of the environment.

Indeed, running a single hook ([run.py#153](https://github.com/pre-commit/pre-commit/blob/49f95b9ef391639d09136666be375d99a3dccd04/pre_commit/commands/run.py#153)) checks against `hook.id` and `hook.alias`, however `install_hook_envs` is invoked with only filtering against `hook.id` ([run.py#423](https://github.com/pre-commit/pre-commit/blob/49f95b9ef391639d09136666be375d99a3dccd04/pre_commit/commands/run.py)

Is this inconsistency deliberate? I couldn't find a use case for it. This quick and dirty patch works for us:

```diff
-        to_install = [hook for hook in hooks if hook.id not in skips]
+        to_install = [hook for hook in hooks if hook.id not in skips and hook.alias not in skips]
```

In our use case Initialization is not needed for everyone (monorepo where some non-python devs also do non-python work) and also very expensive unfortunately (6minutes currently to install already cached pip packages)

(An even better solution would be if initialization wouldn't happen at all if the hook run would we skipped in general (because of `SKIP` variable or because there is no files changed, however, that would be a but bigger patch I suppose))

### pre-commit --version

pre-commit 

### .pre-commit-config.yaml

```yaml
repos:
  - repo: https://github.com/PyCQA/pylint
    rev: v2.14.5
    hooks:
      - name: pylint-alias1
        alias: pylint-alias1
        id: pylint
        additional_dependencies:
          - arrow
      - name: pylint-alias2
        alias: pylint-alias2
        id: pylint
        additional_dependencies:
          - hdfs
```

### ~/.cache/pre-commit/pre-commit.log (if present)

_No response_

### PR DESCRIPTION
**Pull request by kmARC**
Fixes #2478

### CODE REVIEW

File: tests/commands/run_test.py [hunk 1]
```diff
@@ -634,6 +642,13 @@ def test_skip_bypasses_installation(cap_out, store, repo_with_passing_hook):
     )
     assert ret == 0
 
+    ret, printed = _do_run(
+        cap_out, store, repo_with_passing_hook,
+        run_opts(all_files=True),
+        {'SKIP': 'skipme-1'},
+    )
+    assert ret == 1
```
Comment (asottile) [inline tests/commands/run_test.py:650]: this test now validates two separate things -- it should be two separate tests

**Review by asottile — COMMENTED** [review]
just future open source advice, don't make PRs from your `main` branch as it makes it significantly more difficult to review / fix

File: tests/commands/run_test.py (hunk 1 above)
Comment (kmARC) [reply tests/commands/run_test.py:650]: > just future open source advice, don't make PRs from your `main` branch as it makes it significantly more difficult to review / fix

Sure. I was in the gerrit workflow mindset.

I can put it in a separate test method.

I see the CI failing, should I rebase on `v2.20.0` or is this fine?

**Comment by kmARC** [comment]
I am closing this PR and will open one from our company GH account.

**Pull request closed without merging**